lib/ddtrace/tracer.rb

```diff
@@ -288,7 +288,12 @@ def trace(name, options = {})
         # It's not a problem since we re-raise it afterwards so for example a
         # SignalException::Interrupt would still bubble up.
         rescue Exception => e
-          (options[:on_error] || DEFAULT_ON_ERROR).call(span, e)
+          valid_error = (options[:on_error] || DEFAULT_ON_ERROR) ## (options[:on_error] || DEFAULT_ON_ERROR).call(span, e)
+
+          if valid_error.methods.include?(:call)
```

Comment:
You can use a simpler way to verify if the object has a `:call` method: https://ruby-doc.org/core-2.7.2/Object.html#method-i-respond_to-3F

Reply:
@Kyle-Neale in addition to the above comment, we want to be sure that we fallback to `DEFAULT_ON_ERROR` in the case that 1. `options[:on_error]` is `nil` (falsey), or if it does exist (is truthy), that  2. `options[:on_error]` does not respond to `:call`. In either of these cases we want to be sure that we invoke `DEFAULT_ON_ERROR` to at least get the fallback error handling behavior. 

As it's written currently, if `options[:on_error]` is set to something that's incorrect, we won't call any error handling behavior.